model/src/main/kotlin/Repository.kt

```diff
@@ -31,48 +31,51 @@ data class Repository(
     /**
      * Original VCS-related information from the working tree containing the analyzer root.
      */
-    val vcs: VcsInfo,
-
-    /**
-     * Processed VCS-related information from the working tree containing the analyzer root that has e.g. common
-     * mistakes corrected.
-     */
-    val vcsProcessed: VcsInfo = vcs.normalize(),
+    val provenance: KnownProvenance,
 
     /**
      * A map of nested repositories, for example Git submodules or Git-Repo modules. The key is the path to the
      * nested repository relative to the root of the main repository.
      */
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
-    val nestedRepositories: Map<String, VcsInfo> = emptyMap(),
+    val nestedRepositories: Map<String, KnownProvenance> = emptyMap(),
 
     /**
      * The configuration of the repository, parsed from [ORT_REPO_CONFIG_FILENAME].
      */
     val config: RepositoryConfiguration = RepositoryConfiguration()
 ) {
     companion object {
+        private const val DEFAULT_REVISION = ""
+
         /**
          * A constant for a [Repository] where all properties are empty strings.
          */
         @JvmField
         val EMPTY = Repository(
-            vcs = VcsInfo.EMPTY,
-            vcsProcessed = VcsInfo.EMPTY,
+            provenance = RepositoryProvenance(VcsInfo.EMPTY, DEFAULT_REVISION),
             nestedRepositories = emptyMap(),
             config = RepositoryConfiguration()
         )
     }
 
+    fun vcs(): VcsInfo {
```

Comment:
Alternatively, this could be a getter-property annotated with `JsonIgnore`. Same below.

Reply:
Done. Let's see if the tests like me.